### PR TITLE
Add org support for role None

### PR DESCRIPTION
**Pull request by buccarel**
Git card:
https://github.com/grafana/identity-access-team/issues/302

In the UI, the new user looks as follows:
image
Not sure if anything should be done here to display "None" in the "Role" box.

In the DB, the org_user looks as follows:
image

The target version >= 10.2.0 is not yet configured in the CI, so to see that the new unit test is running, I did this locally, against the latest Grafana `main` branch:
```
mattia.buccarella at Mattias-Grafana-Macbook in ~/workspace/terraform-provider-grafana (buccarel/support-role-none)
$ GRAFANA_VERSION=10.2.0 TF_ACC_OSS=true GRAFANA_URL=http://localhost:3000 GRAFANA_AUTH=admin:admin TESTARGS="-run TestAccOrganization_roleNoneUser" make testacc | grep roleNone
TF_ACC=1 go test ./... -v -run TestAccOrganization_roleNoneUser -timeout 120m
=== RUN   TestAccOrganization_roleNoneUser
--- PASS: TestAccOrganization_roleNoneUser (1.82s)

mattia.buccarella at Mattias-Grafana-Macbook in ~/workspace/terraform-provider-grafana (buccarel/support-role-none)
$
```

### CODE REVIEW

File: docs/resources/organization.md
Comment (eleijonmarck) [inline docs/resources/organization.md]:
```suggestion
- `nones` (Set of String) A list of email addresses corresponding to users who should be given the none access to the organization.
```

File: internal/resources/grafana/resource_organization.go
Comment (eleijonmarck) [inline internal/resources/grafana/resource_organization.go]:
```suggestion
A list of email addresses corresponding to users who who should be given none access to the organization.
```

File: internal/resources/grafana/resource_organization_test.go
Comment (julienduchesne) [inline internal/resources/grafana/resource_organization_test.go]: I think this will need a test where there is one in the list

File: docs/resources/organization.md
Comment (julienduchesne) [inline docs/resources/organization.md]: I would call it: `users_without_access` maybe. The others are self descriptive, this one, not so much. A 1:1 match with the API is not required

File: internal/resources/grafana/resource_organization_test.go
Comment (buccarel) [reply internal/resources/grafana/resource_organization_test.go]: Sounds good!
Will do, thanks!

**Comment by eleijonmarck** [comment]
@buccarel @julienduchesne this will be targeting the release of Grafana 10.2. So we have to merge this after the 10.2 release of Grafana.

@julienduchesne could you confirm this to me?

**Comment by eleijonmarck** [comment]
@buccarel on another note, do you know why the "None" role do not show up as a set value to your image.
image

**Comment by buccarel** [comment]
> @buccarel on another note, do you know why the "None" role do not show up as a set value to your image. image

In the description of the PR I guessed that the UI doesn't have the validation of the new role, but in reality I didn't check why this happens. After I finishing addressing all of the comments, I'll try to investigate it.

**Comment by buccarel** [comment]
> @buccarel @julienduchesne this will be targeting the release of Grafana 10.2. So we have to merge this after the 10.2 release of Grafana.
> 
> @julienduchesne could you confirm this to me?

With my understanding of the picture, yes: I already thought this PR should be merged once that version is released.

File: docs/resources/organization.md
Comment (IevaVasiljeva) [inline docs/resources/organization.md]: nit:
```suggestion
- `users_without_access` (Set of String) A list of email addresses corresponding to users who should be given none access to the organization.
```

File: internal/resources/grafana/resource_organization.go
Comment (IevaVasiljeva) [inline internal/resources/grafana/resource_organization.go]:
```suggestion
A list of email addresses corresponding to users who should be given none access to the organization.
```

**Review by IevaVasiljeva — APPROVED** [review]
Changes look good to me! But I agree with you that we need to wait until Grafana 10.2 is released before merging this :)

**Comment by vtorosyan** [comment]
@eleijonmarck pinging you on this one, can you please follow up when capacity allows to ensure we get this change merged as well?

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**
